packages/TorneloScoresheet/src/pages/ResultDisplay/SendEmail.tsx

```diff
@@ -0,0 +1,24 @@
+import qs from 'qs';
+import { Linking } from 'react-native';
+
+export async function sendEmail(to: string, subject: string, body: string) {
+  let url = `mailto:${to}`;
+
+  // Create email link query
+  const query = qs.stringify({
+    subject: subject,
+    body: body,
+  });
+
+  if (query.length) {
+    url += `?${query}`;
+  }
+  // check if we can use this link
+  const canOpen = await Linking.canOpenURL(url);
+
+  if (!canOpen) {
+    throw new Error('Provided URL can not be handled');
+  }
+
+  return Linking.openURL(url);
```

Comment:
I think this function should return a Promise<Result<string>> here, because the linking.openURL will throw an exception if there is a problem, so we need to catch that exception and return a failure so that we can display the error toast

Reply:
Oh wait sorry i just saw the line above it, in our project, we are avoiding throwing exceptions, instead we return a Result object, which can be handled by the client calling this function to display an error